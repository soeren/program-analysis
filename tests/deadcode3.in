// Check for dead code elimination
// - Check that dead code after a  return statement is removed

program

int x;

int toolong(int a){
    int x;
    int y;
    x := a*a;
    return x;
    y := x*42; // dead
    x := y*23; // dead
}

x := toolong(7);

write(x);

end
