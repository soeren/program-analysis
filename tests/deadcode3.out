program
int x;
int toolong(int a) // 3
{
     int x;
     x := a * a; // 4
     return x; // 5
 }
x := toolong(7); // 1
write(x); // 2
end
